### PR TITLE
add a share button to infobox

### LINKED ISSUE #1248: Add a share button to infobox
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**behavior:**

Add a share button like on google

**Screenshots of the issue:**

screenshot_1

**Will work on the issue:**
yes

### PR DESCRIPTION
**Pull request by pythongiant**
add a share button to infobox

Fixes #1248

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:
Add a share button to infobox

-Add a share button to the infobox to share link on twitter/facebook

- https://pr-1255-fossasia-susper.surge.sh
df086500-6543-40f1-9033-21cfbdb10b55

### CODE REVIEW

**Comment by AakashMallik** [comment]
@pythongiant Run this through a linter/prettier. Rest looks good to me.

**Comment by pythongiant** [comment]
@praveenojha33

**Comment by pythongiant** [comment]
@simsausaurabh

**Comment by praveenojha33** [comment]
@pythongiant It can be achieved using CSS.

File: src/app/infobox/infobox.component.html [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+             <h1>Share This Link on {{this.title}}</h1>
+             <hr>
+             <ul>
+                <a href="https://twitter.com/intent/tweet?text=Check%20this%20out%20%23susper%20%40Susper_%20http%3A%2F%2Fsusper.com%2Fsearch%3Fquery%3D{{this.title}}%26start%3D0%26rows%3D10%26mode%3Dtext"> <i class="fa fa-twitter"></i> Tweet about this</a><br>
+                <a href="https://www.facebook.com/sharer.php?u=http://susper.com/search?query={{this.title}}&start=0&rows=10&mode=text"> <i class="fa fa-facebook"></i> Share on Facebook</a>
+             </ul>
+          </div>
+       </div>
+       <button  class="btn dropdown-toggle" type="button" (click)="Show();"><i class="material-icons">share</i></button>
+       {{this.modal.style.display}}
```
[the remark this answers is not in the document]
Comment (AakashMallik) [reply src/app/infobox/infobox.component.html]: @pythongiant I have DMed you the changes required. Do make the changes when you have time.

[the remark this answers is not in the document]
Comment (pythongiant) [reply src/app/infobox/infobox.component.html]: ok thank you

**Comment by praveenojha33** [comment]
Please make rounded corners similar to Google and please also place share button at appropriate place in infobox.

**Comment by pythongiant** [comment]
image
image

**Comment by pythongiant** [comment]
looks good now @praveenojha33 ?

File: src/app/infobox/infobox.component.ts [hunk 2]
```diff
@@ -74,8 +75,26 @@ export class InfoboxComponent implements OnInit {
...
+
+  Show() {
+    this.showModal = true;
+    let modal = document.getElementById("modo");
+    modal.style.display = "block"
+  }
+  close() {
+    this.showModal = false;
+    let modal = document.getElementById("modo");
+    modal.style.display = "none"
```
Comment (AakashMallik) [inline src/app/infobox/infobox.component.ts]: If you are using **ngIf** you don't need this line

File: src/app/infobox/infobox.component.ts [hunk 3]
```diff
@@ -74,8 +75,26 @@ export class InfoboxComponent implements OnInit {
         window.speechSynthesis.cancel();
         window.speechSynthesis.resume();
         window.speechSynthesis.speak(msg);
-      }
+  }
+
+  Show() {
+    this.showModal = true;
+    let modal = document.getElementById("modo");
+    modal.style.display = "block"
```
Comment (AakashMallik) [inline src/app/infobox/infobox.component.ts]: If you are using **ngIf** you don't need this line

File: src/app/infobox/infobox.component.ts [hunk 4]
```diff
@@ -74,8 +75,26 @@ export class InfoboxComponent implements OnInit {
...
+    this.showModal = false;
+    let modal = document.getElementById("modo");
+    modal.style.display = "none"
+  }
 
   ngOnInit() {
+    window.onclick = function(event) {
+      let modal = document.getElementById("modo");
+      if (event.target === modal) {
+        modal.style.display = "none"
```
Comment (AakashMallik) [inline src/app/infobox/infobox.component.ts]: Same as above

**Comment by AakashMallik** [comment]
> image
> image

Copy the CSS from Google's Pop up, It seems a bit off.

File: src/app/infobox/infobox.component.ts (hunk 4 above)
Comment (pythongiant) [reply src/app/infobox/infobox.component.ts]: Yes we do. because for the modal to exit when we click out of the modal

**Comment by pythongiant** [comment]
@AakashMallik I did some changes please review this PR now

**Comment by praveenojha33** [comment]
@pythongiant Good Work, Please decrease the size of heading, when we get sharing options. Also keep equal padding at all sides.

**Comment by pythongiant** [comment]
@praveenojha33 which padding are we talking about?

**Pull request merged**